### ISSUE TITLE
super slow in iPad 3 Retina iOS 9

### ISSUE THREAD

**Issue by TGWC** [adamwulf/loose-leaf#1737]
The draw is super slow, and the undo/redo it's impossible use. Very laggy. Do you know why this happens ?

**Comment by adamwulf** [comment]
not sure - I haven't had any reports of this until now. Is this when you build from source, from the app store, or both?

**Comment by TGWC** [comment]
From the source. It's unusable. I can make a video if you want.

**Comment by adamwulf** [comment]
yeah, a video would be great. Also using the Profiler and seeing where all that time is being spent. Unfortunately I sold my iPad 3 a while ago so I won't be able to repro myself.

**Comment by TGWC** [comment]
I'll upload it later 👍

**Comment by TGWC** [comment]
Here's a quick example, it starts lagging up when I undo.

If I have photos or crops, it stays as the end of the video, super laggy :/

[IMG_0244-1.mp4.zip](https://github.com/adamwulf/loose-leaf/files/817219/IMG_0244-1.mp4.zip)

**Comment by adamwulf** [comment]
I'd be interested in what the Profiler instrument says is going on. I wonder if its maxing out the CPU and lagging b/c of that, or if CPU is relatively low throughout and it's the GPU getting slammed.

If the Profiler doesn't show anything obvious as the bottleneck, the only option might be to swizzle the `[scale]` method on `UIScreen` to return 1 instead of 2. That'd lower the pixels used to 1/4 what it's trying to process now, which I suspect would speed things up considerably. That should be an easy way to get performance up without digging deeper into the root cause. Without access to an iPad 3 I think that's the best advice I can give :/

**Comment by TGWC** [comment]
I deleted         

dispatch_release(sema1);
and so one, because xcode says it's ARC forbids, can be from it ?

**Comment by adamwulf** [comment]
That wouldn't cause it - it's fine to remove those lines. it's something w/ OpenGL, but i can't put my finger on it. i saw something similar ~ a year ago when I was working on iPad Pro support. It's very possibly related to #1551
